app/src/main/java/io/apicurio/registry/services/auth/CustomAuthenticationMechanism.java

```diff
@@ -101,4 +183,11 @@
     public HttpCredentialTransport getCredentialTransport() {
         return new HttpCredentialTransport(HttpCredentialTransport.Type.AUTHORIZATION, "bearer");
     }
+
+    private Uni<SecurityIdentity> authenticateWithClientCredentials(Pair<String, String> clientCredentials, RoutingContext context, IdentityProviderManager identityProviderManager) {
+        OidcAuth oidcAuth = new OidcAuth(httpClient, clientCredentials.getLeft(), clientCredentials.getRight());
```

Comment:
the same about caching. Btw which one of this two are we using in MAS this or `obtainAccessTokenPaswordGrant` ?

Reply:
This one, password grants are not allowed in MAS identity servers.